### ISSUE TITLE
[Bug]: Dropdown and Combox listbox do not have accessible names

### ISSUE THREAD

**Issue by sethreidnz** [microsoft/fluentui#33890]
### Component

Dropdown

### Package version

^9.0.0

### React version

^18

### Environment

```shell
System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  npmPackages:
    @fluentui/react-components: ^9.0.0 => 9.58.3 
    @types/react: ^18 => 18.3.18 
    @types/react-dom: ^18 => 18.3.5 
    react: ^18 => 18.3.1 
    react-dom: ^18 => 18.3.1
```

### Current Behavior

When I test Combox and Drodown using a field as in my stackblitz they listbox has no accessbile name.

### Expected Behavior

The accessbile name should be set to something sensible reflecting the field name.

### Reproduction

https://stackblitz.com/edit/6gz26ktq-y5ffaeub

### Steps to reproduce

1. Open stack blitz
2. See tests running and failing
3. Open example.test.tsx to see what the tests do

### Are you reporting an Accessibility issue?

yes

### Suggested severity

High - No workaround

### Products/sites affected

Affects the examples in storybook so presumably affects any user of fluent v9

### Are you willing to submit a PR to fix?

yes

### Validations

- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [x] The provided reproduction is a minimal reproducible example of the bug.

**Comment by miroslavstastny** [comment]
Technically, you can pass `aria-label` to `listbox` slot:
```tsx
<Dropdown placeholder="Select an animal" listbox={{ "aria-label": "List of animals" }}>
```
@smhigley - should the accessible name be set on the Dropdown's listbox?

**Comment by dmytrokirpa** [comment]
My two cents:

1. We could use what @miroslavstastny suggested, e.g. pass custom `aria-label` to `listbox` slot

2. When the Dropdown/Combobox is used inside a Field with label, we could add the `aria-labeledby="id_of_the_field_label"` to listbox as well


According to [MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/listbox_role):

> [aria-label](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-label)
A human-readable string value which identifies the listbox. If there's a visible label, then [aria-labelledby](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-labelledby) should be used instead to refer to that label.

@smhigley @miroslavstastny wdyt?
